### PR TITLE
Docs for completion

### PR DESCRIPTION
**Pull request by kjeremy**
The first commit adds documentation support to CompletionItems.

The second one I am unsure about. Is that the right way to add docs for functions? If so should I do something similar for other `hir` types and CompletionItems?

### CODE REVIEW

File: crates/ra_hir/src/code_model_api.rs [hunk 1]
```diff
@@ -297,6 +297,7 @@ pub struct FnSignature {
     /// True if the first param is `self`. This is relevant to decide whether this
     /// can be called as a method.
     pub(crate) has_self_param: bool,
+    pub(crate) documentation: String,
```
[the remark this answers is not in the document]
Comment (kjeremy) [reply crates/ra_hir/src/code_model_api.rs]: Should I add a `docs` function or similar to `DefId` then and expose it from`Function`?

File: crates/ra_hir/src/code_model_api.rs [hunk 2]
```diff
@@ -352,6 +352,20 @@ impl Function {
...
         db.generic_params(self.def_id)
     }
+
+    pub fn docs(&self, db: &impl HirDatabase) -> Option<String> {
+        let def_loc = self.def_id.loc(db);
+        let syntax = db.file_item(def_loc.source_item_id);
+        let fn_def = ast::FnDef::cast(&syntax).expect("fn def should point to FnDef node");
+
+        // doc_comment_text unconditionally returns a String
+        let comments = fn_def.doc_comment_text();
```
Comment (matklad) [inline crates/ra_hir/src/code_model_api.rs]: This probably means we should fix `doc_comment_text` :)

Comment (kjeremy) [reply crates/ra_hir/src/code_model_api.rs]: Agreed. I will do that in another PR.

**Comment by kjeremy** [comment]
The test failure appears to be unrelated.

**Comment by matklad** [comment]
It is related. Run `env INSTA_UPDATE=1 cargo test` to update test files.

**Comment by kjeremy** [comment]
Ok should be good.

**Pull request merged**